### PR TITLE
refactor: Refactor action node to declarative style 

### PR DESCRIPTION
**Pull request by Ngkaokis**
ref #2610 #2601

### CODE REVIEW

File: nodes/FormX/actions/document/commonProperties.ts [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+
+		let dataBuffer: Buffer | null = null;
+		if (binaryDataField && this.helpers.assertBinaryData(binaryDataField)) {
+			dataBuffer = await this.helpers.getBinaryDataBuffer(binaryDataField);
+		}
+		requestOptions.body = dataBuffer;
+
+		return requestOptions;
+	} catch (err) {
+		throw new NodeOperationError(this.getNode(), err as Error);
```
Comment (roxk) [inline nodes/FormX/actions/document/commonProperties.ts:23]: What errors would user see in UI?

File: .github/workflows/ci-n8n.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: CI for n8n
+on:
+  workflow_call:
+jobs:
+  ci:
+    name: Run CI for n8n
+    runs-on: ubuntu-latest
```
Comment (roxk) [inline .github/workflows/ci-n8n.yaml:7]: Not using our own runner becoz its public?

**Review by roxk — COMMENTED** [review]
I'm thinking, could we abuse postSend to poll async extract result?

**Comment by Ngkaokis** [comment]
> I'm thinking, could we abuse postSend to poll async extract result?

We do it in declarative way because we want to get merged into n8n repo. I don't think they will approve it if we abuse `postSend` like that :sosad:

File: .github/workflows/ci-n8n.yaml (hunk 2 above)
Comment (Ngkaokis) [reply .github/workflows/ci-n8n.yaml:7]: As we are only running CI here and somehow self hosted runner get disconnected after turning into public, so didn't setup self hosted runner here.

cc @jasonkit

File: nodes/FormX/actions/document/commonProperties.ts (hunk 1 above)
Comment (Ngkaokis) [reply nodes/FormX/actions/document/commonProperties.ts:23]: Actually, I don't see big difference wrapping `NodeOperationError` or not

But seems wrapping it is a standard from what I see in other nodes. (Can't find relevant doc explaining it)
I guess it's for some n8n server internal tracking or sth?

**Comment by roxk** [comment]
@Ngkaokis Could give it a try first? Then we at least have a technical fallback. Whether it works is a technical issue, and whether they approve is a policy issue :hehe: Also, when n8n replies we'd have more info for ten ten to share

File: nodes/FormX/actions/document/commonProperties.ts (hunk 1 above)
Comment (Ngkaokis) [reply nodes/FormX/actions/document/commonProperties.ts:23]: If you mean the error of this function, here is the screenshot. I think it's understandable

image

Comment (roxk) [reply nodes/FormX/actions/document/commonProperties.ts:23]: Hmm I wonder what about other errors, e.g. the field is present but fail to grab the buffer. But I guess for now this suffice. Please keep this in mind first, thanks!

**Comment by Ngkaokis** [comment]
> Could give it a try first? Then we at least have a technical fallback. Whether it works is a technical issue, and whether they approve is a policy issue :hehe: Also, when n8n replies we'd have more info for ten ten to share

Push the hack :sosad:

**Review by roxk — APPROVED** [review]

**Pull request merged**